dsalglib/include/Number_Theory_Algorithms/prime_numbers.h

```diff
@@ -0,0 +1,53 @@
+/*
+ * Created by yadvendra20 on 3/12/2020
+*/
+
+#ifndef DSALGLIB_NUMTHEORY_PRIMES_H
+#define DSALGLIB_NUMTHEORY_PRIMES_H
+
+#include "../array.h"
+
+namespace dsa {
+	
+	// returns true if the number is prime otherwise returns false
+	
+	/*
+	 * The complexity of the isNumPrime function is O(square_root(number))
+	*/
+	
+	bool isNumPrime(int number) {
+		if(number <= 1)
+			return false;
+		for(int i = 2; i * i <= number; i++) {
+			if(number % i == 0)
+				return false;
+		}
+		return true;
+	}
+	
+	// Sieve of eratosthenes
+	
+	/*
+	 * The sieve function returns a boolean array is_prime of 'number' size
+	 * where is_prime[num] = true (if num is a prime number) otherwise false
+	 * 
+	 * More about sieve of eratosthenes: https://cp-algorithms.com/algebra/sieve-of-eratosthenes.html
+	*/
+	
+	array<bool> sieve(long long int number) {
```

Comment:
can this use some sort of optimisation ? making is_prime static or similar other optimisations ?
maybe a object which can be reset with different limit later that again internally just increases length of array